Fix or remove yash-2.7 build

```
/tmp/shell-build.20171216194013.10172 ~/Data/git/shenv/shenv/plugins/shell-build/share/shell-build
/tmp/shell-build.20171216194013.10172/yash-2.7 /tmp/shell-build.20171216194013.10172 ~/Data/git/shenv/shenv/plugins/shell-build/share/shell-build
checking operating system... linux
checking POSIX conformance... 200809
checking SUS conformance... 700
checking whether the compiler works... yes
checking whether the compiler accepts -O2 flag... yes
checking whether system has /proc/self/exe... yes
checking for strnlen... yes
checking for wcsnlen... yes
checking if <sys/stat.h> defines S_ISVTX... yes
checking for eaccess... yes
checking for strsingal... yes
checking for paths.h... yes
checking if getcwd(NULL,0) returns malloc'ed string... yes
checking for getrlimit and setrlimit... yes
checking for RLIMIT_LOCKS... yes
checking for RLIMIT_MEMLOCK... yes
checking for RLIMIT_MSGQUEUE... yes
checking for RLIMIT_NICE... yes
checking for RLIMIT_NPROC... yes
checking for RLIMIT_RSS... yes
checking for RLIMIT_RTPRIO... yes
checking for RLIMIT_SIGPENDING... yes
checking if ctags accepts --recurse option... no
checking if etags accepts --recurse option... no
creating ./Makefile... done
creating builtins/Makefile... done
creating version.h... done
configuration complete
c99 -O2 -g -D HAVE_CONFIG_H  -c arith.c
c99 -O2 -g -D HAVE_CONFIG_H  -c builtin.c
c99 -O2 -g -D HAVE_CONFIG_H  -c exec.c
c99 -O2 -g -D HAVE_CONFIG_H  -c expand.c
c99 -O2 -g -D HAVE_CONFIG_H  -c hashtable.c
c99 -O2 -g -D HAVE_CONFIG_H  -c input.c
c99 -O2 -g -D HAVE_CONFIG_H  -c job.c
c99 -O2 -g -D HAVE_CONFIG_H  -c mail.c
c99 -O2 -g -D HAVE_CONFIG_H  -c option.c
c99 -O2 -g -D HAVE_CONFIG_H  -c parser.c
c99 -O2 -g -D HAVE_CONFIG_H  -c path.c
c99 -O2 -g -D HAVE_CONFIG_H  -c plist.c
c99 -O2 -g -D HAVE_CONFIG_H  -c redir.c
c99 -O2 -g -D HAVE_CONFIG_H  -c mksignum.c
c99 -O2 -g -D HAVE_CONFIG_H  -c strbuf.c
c99 -O2 -g -D HAVE_CONFIG_H  -c util.c
c99 -O2 -g -D HAVE_CONFIG_H  -c variable.c
c99 -O2 -g -D HAVE_CONFIG_H  -c wfnmatch.c
c99 -O2 -g -D HAVE_CONFIG_H  -c yash.c
c99 -O2 -g -D HAVE_CONFIG_H  -c alias.c
c99 -O2 -g -D HAVE_CONFIG_H  -c history.c
make[1]: Entering directory '/tmp/shell-build.20171216194013.10172/yash-2.7/builtins'
c99 -O2 -g -D HAVE_CONFIG_H  -c printf.c
c99 -L/home/pawamoy/.shenv/versions/yash-2.7/lib   -o mksignum mksignum.o -lm
./mksignum > signum.h
c99 -O2 -g -D HAVE_CONFIG_H  -c sig.c
c99 -O2 -g -D HAVE_CONFIG_H  -c test.c
c99 -O2 -g -D HAVE_CONFIG_H  -c ulimit.c
ar -rc builtins.a printf.o test.o ulimit.o
make[1]: Leaving directory '/tmp/shell-build.20171216194013.10172/yash-2.7/builtins'
c99 -L/home/pawamoy/.shenv/versions/yash-2.7/lib   -o yash arith.o builtin.o exec.o expand.o hashtable.o input.o job.o mail.o option.o parser.o path.o plist.o redir.o sig.o strbuf.o util.o variable.o wfnmatch.o yash.o alias.o history.o builtins/builtins.a -lm
Resolving dependencies of source files in "."... done
Resolving dependencies of source files in "builtins"... done
Makefile:54: recipe for target 'depends' failed
make: *** [depends] Error 1
```

--- a/plugins/shell-build/bin/shell-build
+++ b/plugins/shell-build/bin/shell-build
@@ -741,8 +741,40 @@ build_package_standard() {
 }
 
 build_package_standard_build() {
-  build_package_configure "$@"
-  build_package_make "$@"
+  local package_name="$1"
+
+  if [ "${MAKEOPTS+defined}" ]; then
+    MAKE_OPTS="$MAKEOPTS"
+  elif [ -z "${MAKE_OPTS+defined}" ]; then
+    MAKE_OPTS="-j $(num_cpu_cores)"
+  fi
+
+  # Support YAML_CONFIGURE_OPTS, SHELL_CONFIGURE_OPTS, etc.
+  local package_var_name="$(capitalize "${package_name%%-*}")"
+  local PACKAGE_CONFIGURE="${package_var_name}_CONFIGURE"
+  local PACKAGE_PREFIX_PATH="${package_var_name}_PREFIX_PATH"
+  local PACKAGE_CONFIGURE_OPTS="${package_var_name}_CONFIGURE_OPTS"
+  local PACKAGE_CONFIGURE_OPTS_ARRAY="${package_var_name}_CONFIGURE_OPTS_ARRAY[@]"
+  local PACKAGE_MAKE_OPTS="${package_var_name}_MAKE_OPTS"
+  local PACKAGE_MAKE_OPTS_ARRAY="${package_var_name}_MAKE_OPTS_ARRAY[@]"
+  local PACKAGE_CFLAGS="${package_var_name}_CFLAGS"
+
+  if [ "$package_var_name" = "SHELL" ]; then
+      use_homebrew_readline || use_freebsd_pkg ||true
+  fi
+
+  ( if [ "${CFLAGS+defined}" ] || [ "${!PACKAGE_CFLAGS+defined}" ]; then
+      export CFLAGS="$CFLAGS ${!PACKAGE_CFLAGS}"
+    fi
+    if [ -z "$CC" ] && is_mac -ge 1010; then
+      export CC=clang
+    fi
+    ${!PACKAGE_CONFIGURE:-./configure} --prefix="${!PACKAGE_PREFIX_PATH:-$PREFIX_PATH}" \
+      $CONFIGURE_OPTS ${!PACKAGE_CONFIGURE_OPTS} "${!PACKAGE_CONFIGURE_OPTS_ARRAY}" || return 1
+  ) >&4 2>&1
+
+  { "$MAKE" $MAKE_OPTS ${!PACKAGE_MAKE_OPTS} "${!PACKAGE_MAKE_OPTS_ARRAY}"
+  } >&4 2>&1
 }
 
 build_package_standard_install() {
@@ -794,7 +826,7 @@ build_package_make() {
   local PACKAGE_MAKE_OPTS_ARRAY="${package_var_name}_MAKE_OPTS_ARRAY[@]"
 
   if [ "$package_var_name" = "SHELL" ]; then
-      use_homebrew_readline || use_freebsd_pkg || true
+    use_homebrew_readline || use_freebsd_pkg ||true
   fi
 
   { "$MAKE" $MAKE_OPTS ${!PACKAGE_MAKE_OPTS} "${!PACKAGE_MAKE_OPTS_ARRAY}"
@@ -813,22 +845,21 @@ build_package_make_install() {
   } >&4 2>&1
 }
 
-# FIXME: autogen and build_script do exactly the same thing, except for the script name
-build_package_autogen() {
-  local package_name="$1"
-  local package_var_name="$(capitalize "${package_name%%-*}")"
-  local PACKAGE_AUTOGEN="${package_var_name}_AUTOGEN"
-  { chmod +x ${!PACKAGE_AUTOGEN:-./autogen.sh}
-    ${!PACKAGE_AUTOGEN:-./autogen.sh}
-  } >&4 2>&1
-}
-
+# FIXME: build_script and autogen do the same thing, just with a different script name
 build_package_build_script() {
   local package_name="$1"
   local package_var_name="$(capitalize "${package_name%%-*}")"
   local PACKAGE_BUILD_SCRIPT="${package_var_name}_BUILD_SCRIPT"
   { chmod +x ${!PACKAGE_BUILD_SCRIPT:-./Build.sh}
     ${!PACKAGE_BUILD_SCRIPT:-./Build.sh}
+  } >&4 2>&1
+}
+
+build_package_autogen() {
+  local package_name="$1"
+  local package_var_name="$(capitalize "${package_name%%-*}")"
+  local PACKAGE_AUTOGEN="${package_var_name}_AUTOGEN"
+  { ${!PACKAGE_AUTOGEN:-./autogen.sh}
   } >&4 2>&1
 }
 
@@ -1193,7 +1224,7 @@ setup_builtin_patches() {
   local package_patch_path="${DEFINITION_PATH%/*}/patches/${DEFINITION_PATH##*/}/${package_name}"
 
   ORIG_HAS_PATCH="$HAS_PATCH"
-# Apply built-in patches if patch was not given from stdin
+  # Apply built-in patches if patch was not given from stdin
   if [ -z "$HAS_PATCH" ] && [ -d "${package_patch_path}" ]; then
     { find "${package_patch_path}" -maxdepth 1 -type f
     } 2>/dev/null | sort | xargs cat 1>"${package_name}.patch"

--- a/plugins/shell-build/bin/shell-build
+++ b/plugins/shell-build/bin/shell-build
@@ -734,14 +734,31 @@ build_package_warn_unsupported() {
   } >&3
 }
 
-build_package_standard_build() {
-  local package_name="$1"
+# Backword Compatibility for standard function
+build_package_standard() {
+  build_package_standard_build "$@"
+  build_package_standard_install "$@"
+}
 
-  if [ "${MAKEOPTS+defined}" ]; then
-    MAKE_OPTS="$MAKEOPTS"
-  elif [ -z "${MAKE_OPTS+defined}" ]; then
-    MAKE_OPTS="-j $(num_cpu_cores)"
-  fi
+build_package_standard_build() {
+  build_package_configure "$@"
+  build_package_make "$@"
+}
+
+build_package_standard_install() {
+  build_package_make_install "$@"
+}
+
+build_package_standard_install_with_bundled_gems() {
+  { "$MAKE" update-gems
+    "$MAKE" extract-gems
+  } >&4 2>&1
+
+  build_package_standard_install "$@"
+}
+
+build_package_configure() {
+  local package_name="$1"
 
   # Support YAML_CONFIGURE_OPTS, SHELL_CONFIGURE_OPTS, etc.
   local package_var_name="$(capitalize "${package_name%%-*}")"
@@ -749,13 +766,7 @@ build_package_standard_build() {
   local PACKAGE_PREFIX_PATH="${package_var_name}_PREFIX_PATH"
   local PACKAGE_CONFIGURE_OPTS="${package_var_name}_CONFIGURE_OPTS"
   local PACKAGE_CONFIGURE_OPTS_ARRAY="${package_var_name}_CONFIGURE_OPTS_ARRAY[@]"
-  local PACKAGE_MAKE_OPTS="${package_var_name}_MAKE_OPTS"
-  local PACKAGE_MAKE_OPTS_ARRAY="${package_var_name}_MAKE_OPTS_ARRAY[@]"
   local PACKAGE_CFLAGS="${package_var_name}_CFLAGS"
-
-  if [ "$package_var_name" = "SHELL" ]; then
-      use_homebrew_readline || use_freebsd_pkg ||true
-  fi
 
   ( if [ "${CFLAGS+defined}" ] || [ "${!PACKAGE_CFLAGS+defined}" ]; then
       export CFLAGS="$CFLAGS ${!PACKAGE_CFLAGS}"
@@ -766,12 +777,31 @@ build_package_standard_build() {
     ${!PACKAGE_CONFIGURE:-./configure} --prefix="${!PACKAGE_PREFIX_PATH:-$PREFIX_PATH}" \
       $CONFIGURE_OPTS ${!PACKAGE_CONFIGURE_OPTS} "${!PACKAGE_CONFIGURE_OPTS_ARRAY}" || return 1
   ) >&4 2>&1
+}
+
+build_package_make() {
+  local package_name="$1"
+
+  if [ "${MAKEOPTS+defined}" ]; then
+    MAKE_OPTS="$MAKEOPTS"
+  elif [ -z "${MAKE_OPTS+defined}" ]; then
+    MAKE_OPTS="-j $(num_cpu_cores)"
+  fi
+
+  # Support YAML_CONFIGURE_OPTS, SHELL_CONFIGURE_OPTS, etc.
+  local package_var_name="$(capitalize "${package_name%%-*}")"
+  local PACKAGE_MAKE_OPTS="${package_var_name}_MAKE_OPTS"
+  local PACKAGE_MAKE_OPTS_ARRAY="${package_var_name}_MAKE_OPTS_ARRAY[@]"
+
+  if [ "$package_var_name" = "SHELL" ]; then
+      use_homebrew_readline || use_freebsd_pkg || true
+  fi
 
   { "$MAKE" $MAKE_OPTS ${!PACKAGE_MAKE_OPTS} "${!PACKAGE_MAKE_OPTS_ARRAY}"
   } >&4 2>&1
 }
 
-build_package_standard_install() {
+build_package_make_install() {
   local package_name="$1"
   local package_var_name="$(capitalize "${package_name%%-*}")"
 
@@ -783,25 +813,22 @@ build_package_standard_install() {
   } >&4 2>&1
 }
 
-build_package_standard_install_with_bundled_gems() {
-  { "$MAKE" update-gems
-    "$MAKE" extract-gems
-  } >&4 2>&1
-
-  build_package_standard_install "$@"
-}
-
-# Backword Compatibility for standard function
-build_package_standard() {
-  build_package_standard_build "$@"
-  build_package_standard_install "$@"
-}
-
+# FIXME: autogen and build_script do exactly the same thing, except for the script name
 build_package_autogen() {
   local package_name="$1"
   local package_var_name="$(capitalize "${package_name%%-*}")"
   local PACKAGE_AUTOGEN="${package_var_name}_AUTOGEN"
-  { ${!PACKAGE_AUTOGEN:-./autogen.sh}
+  { chmod +x ${!PACKAGE_AUTOGEN:-./autogen.sh}
+    ${!PACKAGE_AUTOGEN:-./autogen.sh}
+  } >&4 2>&1
+}
+
+build_package_build_script() {
+  local package_name="$1"
+  local package_var_name="$(capitalize "${package_name%%-*}")"
+  local PACKAGE_BUILD_SCRIPT="${package_var_name}_BUILD_SCRIPT"
+  { chmod +x ${!PACKAGE_BUILD_SCRIPT:-./Build.sh}
+    ${!PACKAGE_BUILD_SCRIPT:-./Build.sh}
   } >&4 2>&1
 }
 
@@ -827,15 +854,6 @@ build_package_autoconf() {
 
 build_package_autoreconf() {
   { autoreconf --no-recursive
-  } >&4 2>&1
-}
-
-build_package_build_script() {
-  local package_name="$1"
-  local package_var_name="$(capitalize "${package_name%%-*}")"
-  local PACKAGE_BUILD_SCRIPT="${package_var_name}_BUILD_SCRIPT"
-  { chmod +x ${!PACKAGE_BUILD_SCRIPT:-./Build.sh}
-    ${!PACKAGE_BUILD_SCRIPT:-./Build.sh}
   } >&4 2>&1
 }
 

--- a/plugins/shell-build/share/shell-build/yash-2.7
+++ b/plugins/shell-build/share/shell-build/yash-2.7
@@ -1,1 +1,1 @@
-install_package "yash-2.7" "https://osdn.net/projects/yash/downloads/37963/yash-2.7.tar.gz#5ab7e2376681c9c76e37d420c591a5d8665b9bf583b7fcc19e8ba24e7d92ca52"
+install_package "yash-2.7" "https://osdn.net/projects/yash/downloads/37963/yash-2.7.tar.gz#5ab7e2376681c9c76e37d420c591a5d8665b9bf583b7fcc19e8ba24e7d92ca52" configure make_install

